examples/src/test/kotlin/AggregatesBuilderTest.kt

```diff
@@ -0,0 +1,947 @@
+            Aggregates.limit(4)
+            // :snippet-end:
+        ))
+        assertEquals(4, limit.toList().size)
+    }
+
+    @Test
+    fun lookupTest() = runBlocking {
+        data class Comment(@BsonId val id: ObjectId, val name: String, val movie_id: Int, val text: String)
+        data class Results(@BsonId val id: Int, val title: String, val year: Int, val cast: List<String>, val genres: List<String>, val type: String, val rated: String, val plot: String, val fullplot: String, val runtime: Int, val imdb: Movie.IMDB, val joinedComments: List<Comment>)
+        val commentCollection = database.getCollection<Comment>("comments")
+
+        val comments = listOf(
+            Comment(id = ObjectId(), name = "John Doe", movie_id = 1, text = "This is a great movie. I enjoyed it a lot."),
+            Comment(id = ObjectId(), name = "Andrea Le", movie_id = 1, text = "Rem officiis eaque repellendus amet eos doloribus."),
+            Comment(id = ObjectId(), name = "Jane Doe", movie_id = 2, text = "I didn't like this movie. It was boring and predictable."),
+            Comment(id = ObjectId(), name = "Bob Smith", movie_id = 3, text = "This movie is a masterpiece. Highly recommended."))
+        commentCollection.insertMany(comments)
+
+        val lookup = movieCollection.aggregate<Document>(listOf(
+            // :snippet-start: lookup
+            Aggregates.lookup("comments", "_id", Comment::movie_id.name, Results::joinedComments.name
+            // :snippet-end:
+        )))
+        assertEquals(9, lookup.toList().size)
+        assertEquals("The Shawshank Redemption", lookup.toList().first()["title"])
+    }
+
+    @Test
+    fun lookupFullJoinTest() = runBlocking {
+        // :snippet-start: lookup-data-classes
+        data class Order(
+            @BsonId val id: Int,
+            val customerId: Int,
+            val item: String,
+            val ordered: Int
+        )
+        data class Inventory(
+            @BsonId val id: Int,
+            val stockItem: String,
+            val inStock: Int
+        )
+        // :snippet-end:
+        data class StockData(val inStock: Int)
+        data class Results(val item: String, val ordered: Int, val stockData: List<StockData>)
+
+        val orderCollection = database.getCollection<Order>("orders")
+        val warehouseCollection = database.getCollection<Inventory>("warehouses")
+
+        val orders = listOf(Order(1, 1, "item1", 5), Order(2, 1, "item2", 5), Order(3, 1, "item3", 5))
+        val inventory= listOf(Inventory(1, "item1", 10), Inventory(2, "item2", 20), Inventory(3, "item3", 30))
+
+        orderCollection.insertMany(orders)
+        warehouseCollection.insertMany(inventory)
+        // TODO: Confirm why using Filters doesn't populate the stockData field? aren't these equivalent?
+        // Filters.and(
+        //  Filters.eq("order_item", Inventory::stockItem.name),
+        //  Filters.gte(Inventory::inStock.name, "order_qty")
+        //)
```

Comment:
i also couldn't get this example to work with builders. though here's what i think this should be (slightly different from yours):

```kt
Filters.and(
  Filters.eq("\$\$order_item", "\$${Inventory::stockItem.name}"),
  Filters.gte(Inventory::inStock.name, "\$\$order_qty")
)
```

however, using this gives me the following error:

```
Command failed with error 168 (InvalidPipelineOperator): 'Unrecognized expression '$$order_item'' on server ac-j8rabip-shard-00-01.nzumlgv.mongodb.net:27017. The full response is {"ok": 0.0, "errmsg": "Unrecognized expression '$$order_item'", "code": 168, "codeName": "InvalidPipelineOperator", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1684960589, "i": 27}}, "signature": {"hash": {"$binary": {"base64": "LYBjwfW57eKTP5PI8+nGLspt4Sk=", "subType": "00"}}, "keyId": 7200462029784612865}}, "operationTime": {"$timestamp": {"t": 1684960589, "i": 27}}}
com.mongodb.MongoCommandException: Command failed with error 168 (InvalidPipelineOperator): 'Unrecognized expression '$$order_item'' on server ac-j8rabip-shard-00-01.nzumlgv.mongodb.net:27017. The full response is {"ok": 0.0, "errmsg": "Unrecognized expression '$$order_item'", "code": 168, "codeName": "InvalidPipelineOperator", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1684960589, "i": 27}}, "signature": {"hash": {"$binary": {"base64": "LYBjwfW57eKTP5PI8+nGLspt4Sk=", "subType": "00"}}, "keyId": 7200462029784612865}}, "operationTime": {"$timestamp": {"t": 1684960589, "i": 27}}}
	at app//com.mongodb.internal.connection.ProtocolHelper.getCommandFailureException(ProtocolHelper.java:205)
	at app//com.mongodb.internal.connection.InternalStreamConnection.lambda$sendCommandMessageAsync$0(InternalStreamConnection.java:552)
	at app//com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback$MessageCallback.onResult(InternalStreamConnection.java:858)
	at app//com.mongodb.internal.connection.InternalStreamConnection$MessageHeaderCallback$MessageCallback.onResult(InternalStreamConnection.java:821)
	at app//com.mongodb.internal.connection.InternalStreamConnection$3.completed(InternalStreamConnection.java:680)
	at app//com.mongodb.internal.connection.InternalStreamConnection$3.completed(InternalStreamConnection.java:677)
	at app//com.mongodb.internal.connection.AsynchronousChannelStream$BasicCompletionHandler.completed(AsynchronousChannelStream.java:251)
	at app//com.mongodb.internal.connection.AsynchronousChannelStream$BasicCompletionHandler.completed(AsynchronousChannelStream.java:234)
	at app//com.mongodb.internal.connection.tlschannel.async.AsynchronousTlsChannel.lambda$read$4(AsynchronousTlsChannel.java:122)
	at java.base@11.0.12/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base@11.0.12/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base@11.0.12/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base@11.0.12/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base@11.0.12/java.lang.Thread.run(Thread.java:829)
```

curious for tech reviewer's feedback here

Reply:
for now, i think we can just use the Document-based version

Reply:
Yeah this is a complex one. Firstly, have to you `$expr` operator to access the variables from the lookup. Secondly `Filters.eq` & `Filters.gte` don't output the correct MQL for use with `$expr` :(

Reply:
👍 thanks for the explanation. I'll remove the TODO